ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/UnrestrictedPrivacyController.java

```diff
@@ -0,0 +1,287 @@
+    return new PrivacyGroup(privacyGroupId, PrivacyGroup.Type.LEGACY, name, description, addresses);
+  }
+
+  @Override
+  public String deletePrivacyGroup(final String privacyGroupId, final String enclavePublicKey) {
+    throw new RuntimeException("Not Implemented");
+  }
+
+  @Override
+  public PrivacyGroup[] findOffChainPrivacyGroupByMembers(
+      final List<String> addresses, final String enclavePublicKey) {
+    throw new RuntimeException("Not Implemented");
+  }
+
+  @Override
+  public long determineEeaNonce(
+      final String privateFrom,
+      final String[] privateFor,
+      final Address address,
+      final String enclavePublicKey) {
+
+    final String privacyGroupId = createPrivacyGroupId(privateFrom, privateFor);
+    return determineBesuNonce(address, privacyGroupId, enclavePublicKey);
+  }
+
+  private String createPrivacyGroupId(final String privateFrom, final String[] privateFor) {
+    final Bytes32 privacyGroupId =
+        PrivacyGroupUtil.calculateEeaPrivacyGroupId(
+            Bytes.fromBase64String(privateFrom),
+            Arrays.stream(privateFor).map(Bytes::fromBase64String).collect(Collectors.toList()));
+
+    return privacyGroupId.toBase64String();
+  }
+
+  @Override
+  public long determineBesuNonce(
+      final Address sender, final String privacyGroupId, final String enclavePublicKey) {
+    return privateNonceProvider.getNonce(
+        sender, Bytes32.wrap(Bytes.fromBase64String(privacyGroupId)));
+  }
+
+  @Override
+  public Optional<TransactionProcessingResult> simulatePrivateTransaction(
+      final String privacyGroupId,
+      final String enclavePublicKey,
+      final CallParameter callParams,
+      final long blockNumber) {
+    return privateTransactionSimulator.process(privacyGroupId, callParams, blockNumber);
+  }
+
+  @Override
+  public Optional<Bytes> getContractCode(
+      final String privacyGroupId,
+      final Address contractAddress,
+      final Hash blockHash,
+      final String enclavePublicKey) {
+
+    LOG.error(privacyGroupId);
+
+    return privateWorldStateReader.getContractCode(privacyGroupId, blockHash, contractAddress);
+  }
+
+  @Override
+  public PrivateTransactionSimulator getTransactionSimulator() {
+    return privateTransactionSimulator;
+  }
+
+  @Override
+  public Optional<Hash> getStateRootByBlockNumber(
+      final String privacyGroupId, final String enclavePublicKey, final long blockNumber) {
+    return blockchain
+        .getBlockByNumber(blockNumber)
+        .map(
+            block ->
+                privateStateRootResolver.resolveLastStateRoot(
+                    Bytes32.wrap(Bytes.fromBase64String(privacyGroupId)), block.getHash()));
+  }
+
+  @Override
+  public Optional<String> buildAndSendAddPayload(
+      final PrivateTransaction privateTransaction,
+      final Bytes32 privacyGroupId,
+      final String enclaveKey) {
+    throw new RuntimeException("Not Implemented");
```

Comment:
Should this unimplemented methods should include more descriptive message with a custom exception?